### ISSUE TITLE
Free-floating fake tensor without the mode (FakeTensorMode)

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#119589]
### 🐛 Describe the bug

Meta tensors support being allocated on their own, without having a converter or mode associated with them. It is convenient for some use cases for fake tensors to support this too. The FakeTensorMode is primarily needed for maintaining a relationship to real tensors which you have fakeified, but there are valid use cases that don't do this. Fake tensors that don't have mode should be serializable with torch.save.

cc @mruberry @mikaylagawarecki @chauhang @penguinwu @eellison @aorenste @bdhirsh @bobrenjc93 @msaroufim @anijain2305 @zou3519 @suo @thiagocrepaldi 

### Versions

main

**Comment by thiagocrepaldi** [comment]
Is this the complement of #105077 that enabled `torch.load` within faketensormode?

**Comment by eellison** [comment]
Is the real ask here serializing fake tensors ?

**Comment by suo** [comment]
> Is the real ask here serializing fake tensors ?

Yep, that's the context in which it came up.

From discussing with @ezyang like there are two parts to FakeTensor:
- A `(metatensor, device)` pair
- State that maps these FakeTensors to real tensors.

We sometimes want just the `(metatensor, device)` pair—basically any use case that uses metatensors today (e.g. lazy model initialization) would probably be improved by it, since managing device explicitly is good.

One question I have, maybe for @ezyang—besides the real tensor assocation, don't we also stash ShapeEnvs on FakeTensorMode? Does that mean "detached" FakeTensors need to have static shapes?

**Comment by ezyang** [comment]
Yes, detached fake tensors would require static shapes. From my understanding of your requirements, this should be fine.

**Comment by suo** [comment]
yep agreed

**Comment by ezyang** [comment]
Yes, still valid
